src/main/settings.ts

```diff
@@ -98,7 +99,9 @@ export async function saveSettings(settings: Partial<AppSettings>) {
   const currentSettings = await getSettings()
   const newSettings = { ...currentSettings, ...settings }
 
-  newSettings.ai.apiKey = processApiKeyForStorage(newSettings.ai.apiKey)
+  if (currentSettings.ai.apiKey !== newSettings.ai.apiKey) {
```

Comment:
Totally forgot to add this when reviewing the previous PR, but there's a small issue in the Settings form: when you insert a key, it shows the following warning in the console

<img width="1147" height="287" alt="Image" src="https://github.com/user-attachments/assets/52eeccd5-cace-4b25-9b99-9ead515e0c87" />